### PR TITLE
Fix Doctrine deprecations

### LINKED ISSUE #50481: Fix Doctrine deprecations
Since https://github.com/symfony/symfony/pull/50468, we're not blind anymore and we can see deprecations from Doctrine.

Here is an example job that reports many of them:
https://github.com/symfony/symfony/actions/runs/5124419388/jobs/9216226897

I'm calling for help to fix them all, on the lowest branch where applicable since we'd consider fixing those as a bug fix.

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #50481 
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php [hunk 1]
```diff
@@ -480,9 +480,11 @@ private function updateSchema(): void
 
         $schemaManager = $this->createSchemaManager();
         $comparator = $this->createComparator($schemaManager);
-        $schemaDiff = $this->compareSchemas($comparator, $schemaManager->createSchema(), $this->getSchema());
+        $schemaDiff = $this->compareSchemas($comparator, method_exists($schemaManager, 'introspectSchema') ? $schemaManager->introspectSchema() : $schemaManager->createSchema(), $this->getSchema());
```
Comment (brusch) [inline src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php:483]: @nicolas-grekas actually this change deletes all existing tables in the database 😬 
Reverting this code-block fixes the issue.

Comment (xabbuh) [reply src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php:483]: Can you please open a new issue if you think that this change is wrong (best with a failing example application)? Comments on closed PRs are likely to get lost.

Comment (xabbuh) [reply src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php:483]: By the way, reading the code I fail to see how this change can change the actual behaviour (the new `instrospectSchema()` method calls `createSchema()` internally).

Comment (brusch) [reply src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php:483]: @xabbuh yeah, I also thought so, but when I just reverted this code block (not the entire file) it continued to work as expected. But yes, I'll create a follow-up ticket with a reproducible case. Thanks so far 👍😊

**Pull request merged**